DxDispatch/src/dxdispatch/OnnxDispatchable.cpp

```diff
@@ -108,12 +108,33 @@ void OnnxDispatchable::Initialize()
     Ort::ThrowOnError(ortApi.GetExecutionProviderApi("DML", ORT_API_VERSION, reinterpret_cast<const void**>(&ortDmlApi)));
     Ort::ThrowOnError(ortDmlApi->SessionOptionsAppendExecutionProvider_DML1(sessionOptions, m_device->DML(), m_device->GetCommandQueue()));
 
-    m_session = Ort::Session(ortEnvironment, m_desc.sourcePath.wstring().c_str(), sessionOptions);
+    m_session = Ort::Session(*m_environment, m_desc.sourcePath.wstring().c_str(), sessionOptions);
```

Comment:
>Ort::Session [](http://example.com/codeflow?start=16&length=12)

Oof, does the Session really read the `Ort::Env` after this call? IMO, Session should copy all necessary information that it needs right now :/. Otherwise this is a recipe for disaster.

Reply:
Unfortunately, it appears so. I was seeing an AV inside ORT's logging code with an error like this: "Attempt to use the DefaultLogger but none has been registered"

Reply:
-__-